api/src/test/java/com/epam/pipeline/controller/contextual/ContextualPreferenceControllerTest.java

```diff
@@ -0,0 +1,163 @@
+import com.epam.pipeline.entity.contextual.ContextualPreferenceExternalResource;
+import com.epam.pipeline.entity.contextual.ContextualPreferenceLevel;
+import com.epam.pipeline.entity.contextual.ContextualPreferenceSearchRequest;
+import com.epam.pipeline.manager.contextual.ContextualPreferenceApiService;
+import com.epam.pipeline.test.creator.contextual.ContextualPreferenceCreatorUtils;
+import com.epam.pipeline.test.web.AbstractControllerTest;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+
+@WebMvcTest(controllers = ContextualPreferenceController.class)
+public class ContextualPreferenceControllerTest extends AbstractControllerTest {
+
+    private static final String TEST_STRING = "TEST";
+    private static final String CONTEXTUAL_URL = SERVLET_PATH + "/contextual/preference";
+    private static final String LOAD_URL = CONTEXTUAL_URL + "/load";
+    private static final String LOAD_ALL_URL = LOAD_URL + "/all";
+    private static final ContextualPreferenceLevel PREFERENCE_LEVEL = ContextualPreferenceLevel.ROLE;
+    private final ContextualPreference contextualPreference =
+            ContextualPreferenceCreatorUtils.getContextualPreference();
+    private final ContextualPreferenceExternalResource contextualPreferenceExternalResource =
+            ContextualPreferenceCreatorUtils.getCPExternalResource();
+
+    @Autowired
+    private ContextualPreferenceApiService mockContextualPreferenceApiService;
+
+    @Test
+    public void shouldFailLoadAllForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(LOAD_ALL_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAll() throws Exception {
+        final List<ContextualPreference> contextualPreferences = Collections.singletonList(contextualPreference);
+
+        Mockito.doReturn(contextualPreferences).when(mockContextualPreferenceApiService).loadAll();
+
+        final MvcResult mvcResult = performRequest(get(LOAD_ALL_URL));
+
+        Mockito.verify(mockContextualPreferenceApiService).loadAll();
+
+        assertResponse(mvcResult, contextualPreferences,
+                ContextualPreferenceCreatorUtils.CONTEXTUAL_PREFERENCE_LIST_TYPE);
+    }
```

Comment:
I wonder if it is a good idea to group test code lines logically. For instance we could split lines to three parts:
- setup (building resources, configuring mocks)
- test (performing test target operation)
- assert (verifying calls, asserting output)

In this test case it seems to increase readability if the code is organised like this:
```java
    @Test
    @WithMockUser
    public void shouldLoadAll() throws Exception {
        final List<ContextualPreference> contextualPreferences = Collections.singletonList(contextualPreference);
        Mockito.doReturn(contextualPreferences).when(mockContextualPreferenceApiService).loadAll();

        final MvcResult mvcResult = performRequest(get(LOAD_ALL_URL));

        Mockito.verify(mockContextualPreferenceApiService).loadAll();
        assertResponse(mvcResult, contextualPreferences,
                ContextualPreferenceCreatorUtils.CONTEXTUAL_PREFERENCE_LIST_TYPE);
    }
```